impeller/renderer/backend/gles/texture_gles.cc

```diff
@@ -193,7 +193,7 @@ TextureGLES::TextureGLES(std::shared_ptr<ReactorGLES> reactor,
       type_(GetTextureTypeFromDescriptor(GetTextureDescriptor())),
       handle_(external_handle.has_value()
                   ? external_handle.value()
-                  : reactor_->CreateHandle(ToHandleType(type_))),
```

Comment:
I believe textures can be created on any thread. To make the texture object manage its own handle, you'd lazily initialize it in the InitializeContentsIfNecessary function

Reply:
That's what's happening here.  CreateUntrackedHandle creates the texture synchronously.

Reply:
If it creates it synchronously, then we have to be careful not to create texture objects on threads other than raster/IO. but if you defer construction then the existing lifecycle mechanisms manage that for you.

Reply:
It only creates it immediately if CanReactOnCurrentThread is true:

https://github.com/flutter/engine/blob/main/impeller/renderer/backend/gles/reactor_gles.cc#L208-L210

Reply:
it might be getting used by flutter gpu, but if we're not hitting the assert anywhere that is good enough for me